### DISCUSSION TITLE
Permissions required for each construct?

### DISCUSSION THREAD

**Discussion by ZachGerman** [cdklabs/cdk-monitoring-constructs#297] [Q&A] [answered]
If I'm looking to use these to monitor things in another account on my CloudWatch dashboard, I'm guessing I just need to grant the permissions needed for each one to the role invoking the cloudformation create/update stack execution?

**Comment by ghost** [comment]
The library doesn't currently have built-in support for specifying different account IDs for the metrics that are used. We don't have active plans to add it, but PRs are always welcome!

You _could_ use `.monitorCustom()` with arbitrary metrics that include cross-account metrics (with [the necessary permissions](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/Cross-Account-Cross-Region.html), like you mentioned).

**Comment by ZachGerman** [comment, marked as answer]
Yes. It's a bit more typing (because you have to grab all of the metrics in groups), but we figured out how to monitor a cross-account SNS topic using `.monitorCustom()` with `getNewMetric(s1,s2,s3,s4,s5,s6,s7,s8).with({account: <xxxxxxxxxxxx>})`

The CloudWatch permissions stack generated with 'Automatic Dashboards' box checked (CloudWatch -> Settings -> Sharing Configure) took care of everything (I believe this is because we only targeted metrics from the SNS Automatic Dashboards) More permissions may be necessary depending on use-case.

Thanks!
